server.go

```diff
@@ -125,7 +129,6 @@ func NewServer(options ...func(*Server)) *Server {
 	}
 
 	defaultOptions := [...]func(*Server){
-		WithAddr("localhost:9999"),
```

Comment:
There is a problem with this.

Please take a look at the PR I opened in your fork

Reply:
PR merged. I removed WithTLSListener because it added unnecessary complexity. I believe RunTLS and WithListener already provide the user with the necessary customization options. Please correct me if I’m wrong.

Reply:
I thought the same. 👍

Also, semantically, I was annoyed that WithTLSListener was not allowing to pass a listener

Reply:
I'm a bit confused on why this is a problem?